internal/printer/cli.go

```diff
@@ -0,0 +1,76 @@
+package printer
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/crossplane/crossplane/internal/k8s"
+	"github.com/olekukonko/tablewriter"
+)
+
+// Prints out a CLI table of the passed Resource. The fields variable determines the header and values of the table.
+func CliTable(rootResource k8s.Resource, fields []string) error {
```

Comment:
I'm having trouble personally parsing the table view as that format doesn't effectively visually capture the relationships between the resources in the table.

Is it possible to instead pursue a tree view? That would very clearly demonstrate the hierarchy and relationships to the user and overall be more effective. That may be much harder to implement though...what do you think?

Reply:
+1

Reply:
Sure we can try to implement a tree view instead of/ additionally to the table. What do you have in mind for the tree view? Something like kube-lineage, which is a mix of table and tree? Or really only a plain [tree](https://github.com/xlab/treeprint) that instead of file and dir names has the resource information?

Reply:
As discussed in the SIG-DevEx meeting today, a [kube-lineage](https://github.com/tohjustin/kube-lineage) hybrid tree/table view would be ideal, but is probably too difficult to implement before feature freeze.  Let's start with a very simple tree view that shows kind/name with hierarchical indentation and we can always make it more fancy later.

Let's also keep the table view too, no need to throw that work away, but let's make the tree view as the default experience.

Reply:
Added the tree output to the printer package and set it as default. I added a screenshot of how it looks atm. The user also has the option to set the fields as in the other output methods. Please let me know how you like the current design of the output. @phisco @jbw976 

I'll give the kube-lineage-like tree/table a try this weekend but will focus primarily on any feedback for the existing implementation in this PR.